src/mlx/warnings/regex_checker.py

```diff
@@ -130,9 +131,9 @@ def return_check_limits(self):
         '''
         count = 0
         for checker in self.checkers.values():
-            print(f"Counted failures for classification {checker.classification!r}")
-            count += checker.return_check_limits()
-        print(f"total warnings = {count}")
+            padded_string = [f"{string:<20}" for string in [f"{checker.classification}: "]]
+            count += checker.return_check_limits("".join(padded_string))
+        print(f"Returning error code {count}.")
```

Comment:
If the exit code of the tool is 0, it is not an error code. 

```suggestion
        if count:
            print(f"Returning error code {count}.")
```